### PR TITLE
Feature EE-885: Add Core Compatibility in Module Management window and create a process tu publish compatibility versions

### PR DESCRIPTION
**Pull request by tomirodeghiero**

### CODE REVIEW

File: src-db/database/sourcedata/AD_COLUMN.xml [hunk 1]
```diff
@@ -1026,6 +1026,78 @@
 <!--676E5A8C20154C6F900D5B0623150FD2-->  <EM_OBUIAPP_PROCESS_ID><![CDATA[3075F3E3862E459BB28102CE8F26C0DB]]></EM_OBUIAPP_PROCESS_ID>
 <!--676E5A8C20154C6F900D5B0623150FD2--></AD_COLUMN>
 
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9--><AD_COLUMN>
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <AD_COLUMN_ID><![CDATA[6F9FFCAF154B4EEBB9033BC83070C6D9]]></AD_COLUMN_ID>
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <AD_CLIENT_ID><![CDATA[0]]></AD_CLIENT_ID>
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <AD_ORG_ID><![CDATA[0]]></AD_ORG_ID>
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <ISACTIVE><![CDATA[Y]]></ISACTIVE>
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <NAME><![CDATA[Latest_Core]]></NAME>
```
Comment (Matias-Bernal) [inline src-db/database/sourcedata/AD_COLUMN.xml:1034]: Esto deberia coincidir con el name del elemento de sistema

File: src-db/database/sourcedata/AD_COLUMN.xml [hunk 2]
```diff
@@ -1026,6 +1026,78 @@
...
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <COLUMNNAME><![CDATA[Latest_Core]]></COLUMNNAME>
```
Comment (Matias-Bernal) [inline src-db/database/sourcedata/AD_COLUMN.xml:1035]: Faltan la ayuda y descripcion que tiene el elemento

File: src-db/database/sourcedata/AD_COLUMN.xml [hunk 3]
```diff
@@ -1026,6 +1026,78 @@
...
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <ALLOWED_CROSS_ORG_LINK><![CDATA[N]]></ALLOWED_CROSS_ORG_LINK>
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9-->  <IS_CHILD_PROPERTY_IN_PARENT><![CDATA[N]]></IS_CHILD_PROPERTY_IN_PARENT>
+<!--6F9FFCAF154B4EEBB9033BC83070C6D9--></AD_COLUMN>
+
+<!--70ED9EBA09CC45A2B174971B541B2D3A--><AD_COLUMN>
+<!--70ED9EBA09CC45A2B174971B541B2D3A-->  <AD_COLUMN_ID><![CDATA[70ED9EBA09CC45A2B174971B541B2D3A]]></AD_COLUMN_ID>
+<!--70ED9EBA09CC45A2B174971B541B2D3A-->  <AD_CLIENT_ID><![CDATA[0]]></AD_CLIENT_ID>
+<!--70ED9EBA09CC45A2B174971B541B2D3A-->  <AD_ORG_ID><![CDATA[0]]></AD_ORG_ID>
+<!--70ED9EBA09CC45A2B174971B541B2D3A-->  <ISACTIVE><![CDATA[Y]]></ISACTIVE>
+<!--70ED9EBA09CC45A2B174971B541B2D3A-->  <NAME><![CDATA[From_Core]]></NAME>
```
Comment (Matias-Bernal) [inline src-db/database/sourcedata/AD_COLUMN.xml:1070]: idem a lo del otro campo

File: src-db/database/sourcedata/AD_FIELD.xml [hunk 4]
```diff
@@ -81,6 +81,32 @@
 <!--0916ED025F624270B3FE088D699E2775-->  <EM_OBUIAPP_SHOWSUMMARY><![CDATA[N]]></EM_OBUIAPP_SHOWSUMMARY>
 <!--0916ED025F624270B3FE088D699E2775--></AD_FIELD>
 
+<!--10E6BEC6F3444F6BA3C01C2FE9F1D0E8--><AD_FIELD>
+<!--10E6BEC6F3444F6BA3C01C2FE9F1D0E8-->  <AD_FIELD_ID><![CDATA[10E6BEC6F3444F6BA3C01C2FE9F1D0E8]]></AD_FIELD_ID>
+<!--10E6BEC6F3444F6BA3C01C2FE9F1D0E8-->  <AD_CLIENT_ID><![CDATA[0]]></AD_CLIENT_ID>
+<!--10E6BEC6F3444F6BA3C01C2FE9F1D0E8-->  <AD_ORG_ID><![CDATA[0]]></AD_ORG_ID>
+<!--10E6BEC6F3444F6BA3C01C2FE9F1D0E8-->  <ISACTIVE><![CDATA[Y]]></ISACTIVE>
+<!--10E6BEC6F3444F6BA3C01C2FE9F1D0E8-->  <NAME><![CDATA[From Core]]></NAME>
```
Comment (Matias-Bernal) [inline src-db/database/sourcedata/AD_FIELD.xml:89]: falta la descripcion y ayuda

File: src-db/database/sourcedata/AD_FIELD.xml [hunk 5]
```diff
@@ -633,6 +659,32 @@ This process only deletes the selected module, not its dependencies.]]></HELP>
 <!--83681A4AB41B45D9A121D6E5A6EA83AB-->  <EM_OBUIAPP_SHOWSUMMARY><![CDATA[N]]></EM_OBUIAPP_SHOWSUMMARY>
 <!--83681A4AB41B45D9A121D6E5A6EA83AB--></AD_FIELD>
 
+<!--8D7C4BA670C14051A7FBDFEDB340D86F--><AD_FIELD>
+<!--8D7C4BA670C14051A7FBDFEDB340D86F-->  <AD_FIELD_ID><![CDATA[8D7C4BA670C14051A7FBDFEDB340D86F]]></AD_FIELD_ID>
+<!--8D7C4BA670C14051A7FBDFEDB340D86F-->  <AD_CLIENT_ID><![CDATA[0]]></AD_CLIENT_ID>
+<!--8D7C4BA670C14051A7FBDFEDB340D86F-->  <AD_ORG_ID><![CDATA[0]]></AD_ORG_ID>
+<!--8D7C4BA670C14051A7FBDFEDB340D86F-->  <ISACTIVE><![CDATA[Y]]></ISACTIVE>
+<!--8D7C4BA670C14051A7FBDFEDB340D86F-->  <NAME><![CDATA[To Core]]></NAME>
```
Comment (Matias-Bernal) [inline src-db/database/sourcedata/AD_FIELD.xml:667]: idem a lo del otro campo

File: src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java [hunk 6]
```diff
@@ -150,9 +150,9 @@ private void processPackage(Map<String, Object> pkg) throws Exception {
         String name = (String) pkg.get(NAME);
 
         if (!isPackageExcluded(name)) {
-            String[] parts = name.split("\\.");
-            String group = parts[0] + "." + parts[1];
-            String artifact = String.join(".", Arrays.copyOfRange(parts, 2, parts.length));
+        String[] parts = name.split("\\.");
```
Comment (Matias-Bernal) [inline src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java:153]: esto es candidato a conflicto ya que este proceso fue refactorizado

File: src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java [hunk 7]
```diff
@@ -252,7 +252,21 @@ private List<Map<String, Object>> fetchPackageVersions(String packageName) throw
      */
     private void processPackageVersion(Map<String, Object> version, Package pkg) {
         String versionName = (String) version.get(NAME);
-        findOrCreatePackageVersion(pkg, versionName);
+        PackageVersion pkgVersion = findOrCreatePackageVersion(pkg, versionName);
+
+        if (!PackageUtil.ETENDO_CORE.equals(pkg.getArtifact()) && !isPackageExcluded(pkg.getArtifact())) {
```
Comment (Matias-Bernal) [inline src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java:257]: !StringUtils.equals(PackageUtil.ETENDO_CORE,pkg.getArtifact()) && !isPackageExcluded(pkg.getArtifact())...

File: src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java [hunk 8]
```diff
@@ -252,7 +252,21 @@ private List<Map<String, Object>> fetchPackageVersions(String packageName) throw
...
+            String coreVersionRange = PackageUtil.findCoreVersions(pkgVersion.getId());
+
+            if (coreVersionRange != null && !coreVersionRange.isEmpty()) {
```
Comment (Matias-Bernal) [inline src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java:260]: !StringUtils.isEmpty(coreVersionRange)

File: src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java [hunk 9]
```diff
@@ -298,6 +319,13 @@ private PackageVersion findOrCreatePackageVersion(Package pkg, String version) {
             pkgVersion = new PackageVersion();
             pkgVersion.setPackage(pkg);
             pkgVersion.setVersion(version);
+
+            String coreVersionRange = PackageUtil.findCoreVersions(pkgVersion.getId());
+            if (coreVersionRange != null && !coreVersionRange.isEmpty()) {
```
Comment (Matias-Bernal) [inline src/com/etendoerp/dependencymanager/process/GetPackagesFromRepositories.java:324]: !StringUtils.isEmpty(coreVersionRange)

**Comment by Gremiger** [comment]
@tomyrodeghiero check Matias comments. If the PR is not ready to be merged when you do the MR, but you waant it to be here in github, you can upload it as a Draft PR. (@facumoyanosmf can you check if the PR Activity bot excludes the Draft MR from github? thanks!)

image

**Comment by tomirodeghiero** [comment]
> @tomyrodeghiero check Matias comments. If the PR is not ready to be merged when you do the MR, but you waant it to be here in github, you can upload it as a Draft PR. (@facumoyanosmf can you check if the PR Activity bot excludes the Draft MR from github? thanks!)
> 
> image

@Gremiger Thank you for the comment! I decided to close this pull request until I finish correcting the changes and implementing what is required

**Pull request closed without merging**